### PR TITLE
[4] Avatar Entities

### PR DESCRIPTION
**Pull request by IDragonfire**
DB PR https://github.com/FAForever/db/pull/78

### CODE REVIEW

File: src/test/java/com/faforever/integration/factories/AvatarFactory.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import lombok.Builder;
+import org.springframework.util.Assert;
+
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+
+public class AvatarFactory {
+  public static final String DEFAULT_TOOLTIP = "Cool Test Avatar Tooltip";
+  public static final String DEFAULT_URL = "www.faf-test.org/myavatar.png";
```
Comment (micheljung) [inline src/test/java/com/faforever/integration/factories/AvatarFactory.java]: That's not an URL :) URLs always start with `http://` (or https, of course)

File: src/test/java/com/faforever/integration/factories/AvatarFactory.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  @Builder
+  public static Avatar createAvatar(String tooltip, String url,
+                                    Player player,
+                                    TestDatabase database) {
```
Comment (micheljung) [inline src/test/java/com/faforever/integration/factories/AvatarFactory.java]: Join lines

File: src/test/java/com/faforever/integration/factories/PlayerFactory.java [hunk 3]
```diff
@@ -3,14 +3,26 @@
...
+import static org.junit.Assert.assertEquals;
 
 public class PlayerFactory {
-  public static Player createPlayer(String login, TestDatabase database) throws Exception {
+  private static final String DEFAULT_USERNAME = "JUnitTestUser_PlayerFactory";
+  private static final String DEFAULT_PASSWORD = "testPassword";
+
+
+  @Builder
+  private static Player create(String login, String password, TestDatabase database) throws Exception {
```
Comment (micheljung) [inline src/test/java/com/faforever/integration/factories/PlayerFactory.java]: Codacy says this is unused?

Comment (IDragonfire) [inline src/test/java/com/faforever/integration/factories/PlayerFactory.java]: Codacy has problems with the `@Builder` from lombok ...
image

File: src/main/java/com/faforever/api/data/domain/AvatarAssignment.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.time.OffsetDateTime;
+
+@Entity
+@Table(name = "avatars")
+@Include(rootLevel = true, type = "avatarAssignment")
+@Setter
+public class AvatarAssignment {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/data/domain/AvatarAssignment.java]: Shouldn't use AbstractEntity?

Comment (IDragonfire) [inline src/main/java/com/faforever/api/data/domain/AvatarAssignment.java]: yes

**Comment by micheljung** [comment]
Looks good, can you solve the conflicts so I can merge?

**Comment by IDragonfire** [comment]
@micheljung ready for merge, ignore codacy ...

**Pull request merged**